src/core/api/schema.py

```diff
@@ -100,6 +103,31 @@ class JobBase(HashableBaseModel):
     location: str
 
 
+class JobStat(BaseModel):
+    """Base class for Jobs that need stats."""
+
+    total_objects: int
+    total_labels: int
+    labels: Dict[str, int] = dict()
+
+    @validator("labels", pre=False)
```

Comment:
Did you test `each_item` or `whole` on validator? I see some bug report on this in the past.
```suggestion
    @validator("labels", pre=False, each_item=True)
```

Reply:
According to the docs, that's needed when you aren't iterating over the list you are validating. Ref: https://pydantic-docs.helpmanual.io/usage/validators/#subclass-validators-and-each_item

Since I am iterating over the data structure, it shouldn't be needed.

Reply:
I also can't find any docs for `whole`?

Reply:
Maybe whole has been chnaged, but it is mentioned here when validators is running twice:

https://github.com/samuelcolvin/pydantic/issues/737
https://github.com/samuelcolvin/pydantic/issues/254

Reply:
I dug around some more. `whole=True` is literally `each_item=False` (looked at the code). Which is default behavior, and does not fix the issue...

`each_item=True` is only providing the values, so I can't use that as a workaround either.

This also only happens when running through FastAPI. Simply creating a `JobStat` with the default constructor or `from_orm()` is working as expected.

I also think this is a result of us perhaps miss using the validators, using them to *translate* rather than *validating*. The current code is technically translating and validating, so it's technically okay, though somewhat messy.